CMakeLists.txt

```diff
@@ -66,6 +66,14 @@ if(NOT MSVC)
     add_compile_options(-fno-omit-frame-pointer)
 endif()
 
+include(FetchContent)
+
+FetchContent_Declare(cmake_git_version_tracking                   
+  GIT_REPOSITORY https://github.com/andrew-hardin/cmake-git-version-tracking.git
+  GIT_TAG 904dbda1336ba4b9a1415a68d5f203f576b696bb
+)
+FetchContent_MakeAvailable(cmake_git_version_tracking)
+
```

Comment:
Can you try using a submodule instead? This is for consistency with how all other libraries are added. It can be put it in the 3rdparty directory, and added using add_subdirectory. the other dependencies work as examples.

Reply:
Something I'm curious/concerned about is how this library selects which git repository to read info from. Most likely, it won't select itself. It somehow has to obtain the parent that added it.

Reply:
Alright I'll do some testing and see how that works out